src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h

```diff
@@ -0,0 +1,155 @@
+// $Maintainer: Douglas McCloskey $
+// $Authors: Douglas McCloskey $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_ANALYSIS_OPENSWATH_MRMFEATUREQC_H
+#define OPENMS_ANALYSIS_OPENSWATH_MRMFEATUREQC_H
+
+#include <OpenMS/KERNEL/MRMFeature.h>
+#include <OpenMS/KERNEL/Feature.h>
+#include <OpenMS/KERNEL/FeatureMap.h>
+#include <OpenMS/FORMAT/QcMLFile.h>
+
+#include <OpenMS/DATASTRUCTURES/String.h>
+#include <OpenMS/CONCEPT/LogStream.h>
+
+namespace OpenMS
+{
+
+  /**
+
+    @brief The MRMFeatureQC is a class to handle the parameters and options for 
+      MRMFeatureFilter. The format is based on the TraML format.
+
+  */
+  class OPENMS_DLLAPI MRMFeatureQC
+  {
+
+public:
+
+    //@{
+    /// Constructor
+    MRMFeatureQC();
+
+    /// Destructor
+    ~MRMFeatureQC();
+    //@}
+
+    // Members
+
+    /// QCs within a component group
+    struct ComponentGroupQCs
+    {
+      /// name of the component group
+      String component_group_name_; 
+
+      // number of transitions and labels
+      /// number of heavy ion lower bound
+      int n_heavy_l_;
+      /// number of heavy ion upper bound
+      int n_heavy_u_;
+      int n_light_l_;
+      int n_light_u_;
+      int n_detecting_l_;
+      int n_detecting_u_;
+      int n_quantifying_l_;
+      int n_quantifying_u_;
+      int n_identifying_l_;
+      int n_identifying_u_;
+      int n_transitions_l_;
+      int n_transitions_u_;
+
+      // Ion Ratio QCs
+      String ion_ratio_pair_name_1_;
+      String ion_ratio_pair_name_2_;
+      double ion_ratio_l_;
+      double ion_ratio_u_;
+      String ion_ratio_feature_name_;
+        
+    };
+
+    // QCs for individual components
+    struct ComponentQCs
+    {
+      /// name of the component
+      String component_name_; 
+
+      // Feature members
+      /// retention time lower bound
+      double retention_time_l_;
+      /// retention time upper bound
+      double retention_time_u_;
+      /// intensity lower bound
+      double intensity_l_;
+      /// intensity upper bound
+      double intensity_u_;
+      /// overall quality lower bound
+      double overall_quality_l_;
+      /// overall quality upper bound
+      double overall_quality_u_;
+
+      /// Feature MetaValues
+      std::map<String,std::pair<double,double>> meta_value_qc_;
+
+    };
+
+    // QCs for multiple components (between or within component_groups)
+    struct ComponentGroupPairQCs
+    {
+
+      /// name of the component
+      String component_group_name_; 
+      /// name of the component to calculate the resolution or retention time 
+      String resolution_pair_name_; 
+      /// resolution lower bound 
+      double resolution_l_;
+      /// resolution upper bound 
+      double resolution_u_;
+      /// retention time lower bound 
+      double rt_diff_l_;
+      /// retention time upper bound 
+      double rt_diff_u_;
+    };
+
```

Comment:
can you make the members private or do you need to access them?

if they are public, please dont use a final underscore, eg use `components_qcs` instead

Reply:
They are public.  I will go ahead and change that.